src/html2pdf.js

```diff
@@ -32,356 +32,412 @@
+            overlay.appendChild(container);
+            document.body.appendChild(overlay);
+
+            // Enable page-breaks.
+            var pageBreaks = source.querySelectorAll('.html2pdf__page-break');
+            var pxPageHeight = pageSize.inner.height * pageSize.k / 72 * 96;
+            Array.prototype.forEach.call(pageBreaks, function (el) {
+                el.style.display = 'block';
+                var clientRect = el.getBoundingClientRect();
+                el.style.height = pxPageHeight - (clientRect.top % pxPageHeight) + 'px';
+            }, this);
+
+            // Return the container.
+            return container;
+        };
+
+        html2pdf.makePDF = function (canvas, pageSize, opt) {
+            // Calculate the number of pages.
+            var ctx = canvas.getContext('2d');
+            var pxFullHeight = canvas.height;
+            var pxPageHeight = Math.floor(canvas.width * pageSize.inner.ratio);
+            var nPages = Math.ceil(pxFullHeight / pxPageHeight);
+            var pageModulus = pxFullHeight % pxPageHeight;
+
+            // Create a one-page canvas to split up the full image.
+            var pageCanvas = document.createElement('canvas');
+            var pageCtx = pageCanvas.getContext('2d');
+            var pageHeight = pageSize.inner.height;
+            pageCanvas.width = canvas.width;
+            pageCanvas.height = pxPageHeight;
+
+            // Initialize the PDF.
+            var pdf = new jsPDF(opt.jsPDF);
+
+            /**
+             * added support for page split only on white "rows".
+             * check if pixel is not "white" if it is less than whiteColorLevel value for the rgb values
+             * if the row isn't "white" - will go one pixel up and test that row, up to max. number of maxPixelToGoUp rows up
+             *
+             */
+            var whiteColorLevel = 150;
+            var maxPixelToGoUp = 150;
+            var pageStartPos = 0;
+            for (var page = 0; page < nPages; page++) {
+                // Trim the final page to reduce file size.
+                if (page === nPages - 1) {
+                    pageCanvas.height = pageModulus;
+                    pageHeight = pageCanvas.height * pageSize.inner.width / pageCanvas.width;
+                }
+                // Display the page.
+                var w = pageCanvas.width;
+                var heightDrawBack = 0;
+                do {
+
+                    var h = pageCanvas.height - heightDrawBack;
+                    var whiteRow = true;
+                    var wPix = 0;
+                    var bPixCount = 0;
+
+                    while (wPix < w && whiteRow) {
+                        var pixelData = ctx.getImageData(wPix, pageStartPos + h, 1, 1).data;
+                        if (pixelData[0] < whiteColorLevel || pixelData[1] < whiteColorLevel || pixelData[2] < whiteColorLevel) {
+                            bPixCount++;
+                            if (bPixCount > 1) {
```

Comment:
What's the motivation for allowing up to 1 pixel to have 1 RGB value below the threshold?

Reply:
for some reason - even though my page is with white bg, the getImageData function always thinks that the last pixel on the row is black, so that one is special case for me.
maybe add parameter of how much "not white" pixel one allow to have on a row? (for example - if one have border in his page - he will allow 2 pixels)